internal/caching/impl_inmemorylru.go

```diff
@@ -83,6 +89,15 @@ func NewInMemoryLRUCache(enablePrometheus bool) (*Caches, error) {
 	}, nil
 }
 
+func cacheCleaner(caches ...*InMemoryLRUCachePartition) {
+	for {
+		time.Sleep(time.Minute)
+		for _, cache := range caches {
+			cache.lru.RemoveOldest()
```

Comment:
It would be nice to keep a minimum number of items in the cache, else I fear a small quiet time may result in lots of CPU churn as we re-populate the cache.